src/clj/game/cards/upgrades.clj

```diff
@@ -103,6 +103,21 @@
                                       :unsuccessful {:effect (effect (system-msg "trashes Bernice Mai from the unsuccessful trace")
                                                                      (trash card))}}}}}
 
+  "Bio Vault"
+  {:implementation "Installation restriction not enforced"
+   :advanceable :always
+   :abilities [{:label "[Trash], 2 hosted advancement tokens: End the run."
+                :advance-counter-cost 2
+                :req (req (:run @state))
+                :msg "end the run. Bio Vault is trashed"
+                :delayed-completion true
+                :effect (effect (continue-ability
+                                  {:delayed-completion true
+                                   :effect (effect
+                                             (end-run)
+                                             (trash eid card {:cause :ability-cost}))}
```

Comment:
I think this can all be in the root `:effect`, without the continue-ability.

Reply:
@nealterrell not quite sure why I nested it like that in the first place. I think I must have thought that the `eid` being passed to `trash` needed a `continue-` or `resolve-ability`.